pandas/tests/io/parser/test_na_values.py

```diff
@@ -669,3 +676,47 @@ def test_bool_and_nan_to_float(all_parsers):
     result = parser.read_csv(StringIO(data), dtype="float")
     expected = DataFrame.from_dict({"0": [np.nan, 1.0, 0.0]})
     tm.assert_frame_equal(result, expected)
+
+
+@skip_pyarrow
+def test_boolean_na_values(all_parsers):
```

Comment:
Pandas does typically not add tests for type annotations (pandas-stubs does https://github.com/pandas-dev/pandas-stubs/). Most of the pandas tests are not checked by mypy (and pyright).